### DISCUSSION TITLE
Welcome to MissionControl Discussions!

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#159] [General]
## 👋 Welcome!
  I'm trying out the new Github discussions feature as a place for users to engage in extended or less technical discussions, and save the issue tracker for bug reports and other technical issues as it was intended. This space is intended for things like:
  * Extended discussions that may originate from issue topics.
  * Asking for support as a user.
  * Asking questions you’re wondering about.
  * Giving feedback and sharing ideas.
  * Engaging with other community members.

While I will be more lenient with discussions here, please try to keep conversations relatively on-topic. Any discussion of piracy or piracy enabling tools is strictly forbidden.

**Comment by impeeza** [comment]
👏👏👏👏👏

**Reply by leocruga** [reply]
Great developers! Would it be possible to add support for the Gamesir T1s Control? it would be of great value, thank you very much!

**Comment by Velvis70** [comment]
Hi,
I was wondering what the estimated timeline is for Bluetooth LE devices. I bought one of these 
https://www.icode.com/product/bluetooth-wireless-atari-retro-joystick-and-paddle-adapter/
to use Atari Paddles as a USB device for emulation but it shows up in Windows as a LE device.

A follow up question is it possible to sync a device without a rumble feature? The Change Grip/Order seems to indicate you can only sync devices by activating the vibration feature.

Looking forward to hearing from you and future releases.

**Reply by ndeadly** [reply]
Hey, thanks for trying out the discussions section! In the future you should open a new discussion to ask your questions, as this is the comments for the welcome message.

I don't really have an estimate for BLE support, as it's not really a straightforward development task and requires more reverse engineering and solutions to a few major issues before I can begin to implement it. I have a proof of concept working where I can manually connect controllers and receive data from them, but because the console has no functions for handling generic HID controllers over BLE it is unable to manage them once connected. The challenge will be writing code to handle them while making the other system components aware of their presence and not breaking anything in the process. I'd say we're probably at least several months away from having something stable working.

I'm not sure where you got the idea that only devices supporting rumble can be paired from that screen, but no, that's not the case. You can currently pair almost any device that uses classic Bluetooth. Unsupported controllers will need handlers created before they will work, but they should still pair and connect. Such controllers are distinguished by appearing in the UI with red buttons to indicate we aren't handling the input.

**Comment by TheOneEyedGrimReaper** [comment]
Yo!
Thanks for your work man.
This is f*king awesome to use ds4 and ds5 in docked mode, my lil bro is killed my left joycon so i was unable to play wirelessly in docked mode.
i hope the gyro support is coming soon, it would give some plus to some games thats using it.

**Comment by Musi95** [comment]
Is there a way to rewrite or configure the buttons

**Reply by ndeadly** [reply]
Use the native button remapping feature introduced to HOS in firmware 10.0.0

**Reply by Musi95** [reply]
> Use the native button remapping feature introduced to HOS in firmware 10.0 0

Idk why I didn't think of that. I think you're talking about the button mapping in controllers in settings correct me if I'm wrong

**Reply by ndeadly** [reply]
Yeah that's correct. It will work for all controllers supported by mission control, since they emulate switch pro controllers. Much easier to rely on this built-in feature than to reinvent the wheel and add additional complexity and bloat to mission control. There are a few limitations (eg can't do things like switch dpad with analog stick, invert sticks etc) that I will probably address at some point though.

For most controllers it should be pretty intuitive about what the default button mappings used by mission control are. For other less standard controllers like the wiimote and its extensions you can refer to the source code to see what they use https://github.com/ndeadly/MissionControl/tree/master/bluetooth-mitm/source/controllers

**Reply by Musi95** [reply]
> Yeah that's correct. It will work for all controllers supported by mission control, since they emulate switch pro controllers. Much easier to rely on this built-in feature than to reinvent the wheel and add additional complexity and bloat to mission control. There are a few limitations (eg can't do things like switch dpad with analog stick, invert sticks etc) that I will probably address at some point though.
> 
> For most controllers it should be pretty intuitive about what the default button mappings used by mission control are. For other less standard controllers like the wiimote and its extensions you can refer to the source code to see what they use https://github.com/ndeadly/MissionControl/tree/master/bluetooth-mitm/source/controllers

Awesome, thanks for the help!
